web-registry/src/components/organisms/ChooseCreditClass/hooks/useGetCreditClassOptions.tsx

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect } from 'react';
+
+import useQueryListClassesWithMetadata from '../../../../hooks/useQueryListClassesWithMetadata';
+import { client } from '../../../../sanity';
+import { useAllCreditClassQuery } from '../../../../generated/sanity-graphql';
+import { useAllCreditClassesQuery } from '../../../../generated/graphql';
+import { useWallet } from '../../../../lib/wallet';
+import { CreditClassMetadataLD } from '../../../../generated/json-ld';
+
+interface CreditClassOption {
+  id: string;
+  onChainId: string;
+  imageSrc?: string;
+  title: string;
+  description?: string;
+  metadata: Partial<CreditClassMetadataLD>;
+}
+
+function useGetCreditClassOptions(basketDenom?: string): CreditClassOption[] {
+  const { wallet } = useWallet();
+  const [creditClassOptions, setCreditClassOptions] = useState<
+    CreditClassOption[]
+  >([]);
+
+  const onChainClasses = useQueryListClassesWithMetadata();
+  const { data: offChainCreditClasses } = useAllCreditClassesQuery();
+  const { data: creditClassContentData } = useAllCreditClassQuery({ client });
+
+  useEffect(() => {
+    if (
+      wallet?.address &&
+      onChainClasses &&
+      onChainClasses?.length > 0 &&
+      creditClassContentData
+    ) {
+      const onChainCreditClasses = onChainClasses?.filter(
+        onChain => onChain.issuers?.indexOf(wallet?.address || '') > -1,
+      );
+      const offChainClasses =
+        onChainCreditClasses && onChainCreditClasses?.length > 1
+          ? offChainCreditClasses?.allCreditClasses?.nodes?.filter(
+              offChain =>
+                offChain?.onChainId &&
+                onChainCreditClasses?.findIndex(
+                  onChain => onChain.classId === offChain.onChainId,
+                ) > -1,
+            )
+          : [];
+
+      if (offChainClasses && offChainClasses?.length > 0) {
```

Comment:
Should we display credit classes that the user is not an issuer of? The ticket says:

> Only show credit classes for which the current user wallet address is part of the issuers list 

But we could show them all, and then give the user an error if they are not an issuer?

@erikalogie @S4mmyb

Reply:
Yeah I think that would be good @mhagel. We should be showing all on-chain credit classes, but only allow them to create projects for credit classes they are an issuer for. I'd say just do whatever is easiest then in the future we could explore updating the designs to allow it to either throw an error or just gray out those they aren't an issuer for? Thoughts @erikalogie?

Reply:
I think graying those out would be best, with some tooltip saying "you can't create projects for credit class you're not an issuer of" for instance

Reply:
but maybe just displaying some error banner would be more straightforward to implement for now and we can work on a better version in the future

Reply:
@mhagel since we now want to display all the on chain credit class, this means that you won't be able to provide the off-chain creditClassId to the updateProject mutation
but we should still save the selected credit class somehow in the project so that we can eventually submit it on chain at the end of the flow, so probably good to add a field in the metadata to store the on chain credit class id that got selected

Reply:
I like that idea @blushi. I made some design for this: https://www.figma.com/file/TyhFaXqko9F8OvLEE4Zomg/Issuer-Create%2FEdit-Projects%2C-Credit-Class%2C-Batches?node-id=2%3A4630. I think we could leave this as-is for now and implement the greyed out credit class cards as part of a follow up task.